src/bootstrap/compile.rs

```diff
@@ -757,6 +759,9 @@ pub fn rustc_cargo(builder: &Builder<'_>, cargo: &mut Cargo, target: TargetSelec
         .arg(builder.rustc_features(builder.kind))
         .arg("--manifest-path")
         .arg(builder.src.join("compiler/rustc/Cargo.toml"));
+
+    cargo.rustdocflag("-Zcrate-attr=deny(elided_lifetimes_in_paths)");
```

Comment:
I'm not sure this has the intended effect? I think if you just make this change and revert the changes to doctests, you won't see any warnings; the attribute is being applied to the crate itself, not to the doctests in it. See my comment above about doc(test(warn(...))).

Reply:
I thought this attribute will be enough because it's under the `rustdocflag`. Also, it works as expected.

### Here is the demonstration:

Diff:
```diff
diff --git a/compiler/rustc_graphviz/src/lib.rs b/compiler/rustc_graphviz/src/lib.rs
index 5d86d895817..549cf4bf850 100644
--- a/compiler/rustc_graphviz/src/lib.rs
+++ b/compiler/rustc_graphviz/src/lib.rs
@@ -164,7 +164,7 @@
 //!     fn node_id(&'a self, n: &Nd) -> dot::Id<'a> {
 //!         dot::Id::new(format!("N{}", n)).unwrap()
 //!     }
-//!     fn node_label(&self, n: &Nd) -> dot::LabelText<'_> {
+//!     fn node_label(&self, n: &Nd) -> dot::LabelText {
 //!         dot::LabelText::LabelStr(self.nodes[*n].into())
 //!     }
 //!     fn edge_label(&self, _: &Ed<'_>) -> dot::LabelText<'_> {
@@ -177,7 +177,7 @@
 //!     type Edge = Ed<'a>;
 //!     fn nodes(&self) -> dot::Nodes<'a,Nd> { (0..self.nodes.len()).collect() }
 //!     fn edges(&'a self) -> dot::Edges<'a,Ed<'a>> { self.edges.iter().collect() }
-//!     fn source(&self, e: &Ed<'_>) -> Nd { let & &(s,_) = e; s }
+//!     fn source(&self, e: &Ed) -> Nd { let & &(s,_) = e; s }
 //!     fn target(&self, e: &Ed<'_>) -> Nd { let & &(_,t) = e; t }
 //! }
 //!
```

Part of the `x test compiler/rustc_graphviz` output:
```sh
running 6 tests
...F..
failures:

---- src/lib.rs - (line 142) stdout ----
error: hidden lifetime parameters in types are deprecated
  --> src/lib.rs:167:42
   |
28 |     fn node_label(&self, n: &Nd) -> dot::LabelText {
   |                                     -----^^^^^^^^^
   |                                     |
   |                                     expected lifetime parameter
   |
note: the lint level is defined here
  --> <crate attribute>:1:6
   |
1  | deny(elided_lifetimes_in_paths)
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^
help: indicate the anonymous lifetime
   |
28 |     fn node_label(&self, n: &Nd) -> dot::LabelText<'_> {
   |                                                   ++++

error: hidden lifetime parameters in types are deprecated
  --> src/lib.rs:180:26
   |
41 |     fn source(&self, e: &Ed) -> Nd { let & &(s,_) = e; s }
   |                          ^^ expected lifetime parameter
   |
help: indicate the anonymous lifetime
   |
41 |     fn source(&self, e: &Ed<'_>) -> Nd { let & &(s,_) = e; s }
   |                            ++++

error: aborting due to 2 previous errors

Couldn't compile the test.

failures:
    src/lib.rs - (line 142)

test result: FAILED. 5 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.22s
```

Reply:
> Ok, no need to change it, but maybe leave a comment in builder.rs explaining what's going on?

Sure.

What do you think this diff?
```diff
diff --git a/src/bootstrap/builder.rs b/src/bootstrap/builder.rs
index b54bf432621..34304e4854c 100644
--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -2048,6 +2048,10 @@ pub(crate) fn open_in_browser(&self, path: impl AsRef<Path>) {
 mod tests;

 #[derive(Debug, Clone)]
+/// Represents flag values in `String` form with whitespace delimiter to pass it to the compiler later.
+///
+/// `-Z crate-attr` flags will be applied recursively on the target code using the `rustc_parse::parser::Parser`.
+/// See `rustc_builtin_macros::cmdline_attrs::inject` for more information.
 struct Rustflags(String, TargetSelection);

 impl Rustflags {
```

Reply:
That works :) I was imagining a comment where you added crate-attr but this is good too.

Reply:
> maybe leave a comment in builder.rs explaining what's going on?

I guess this was a typo then, `crate-attr` is added in `compile.rs` :)